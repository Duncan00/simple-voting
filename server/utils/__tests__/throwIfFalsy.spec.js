'use strict';

const throwIfFalsy = require('../throwIfFalsy');

describe('throwIfFalsy', () => {
	test('should throw if not object', () => {
		expect(() => {
			throwIfFalsy('not an object');
		}).toThrowErrorMatchingSnapshot('error snapshot - not an object');
	});

	test('should throw if object contains falsy values', () => {
		const obj = {
			valid: 'hello world!',
			invalid: null
		};

		expect(() => {
			throwIfFalsy(obj);
		}).toThrowErrorMatchingSnapshot(
			'error snapshot - object contains falsy'
		);
	});
});
