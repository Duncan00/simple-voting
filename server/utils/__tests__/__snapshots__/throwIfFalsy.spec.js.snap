// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throwIfFalsy should throw if not object: error snapshot - not an object 1`] = `"please wrap you param in a {} so that the function knows the var name by object key"`;

exports[`throwIfFalsy should throw if object contains falsy values: error snapshot - object contains falsy 1`] = `"variable is falsy: invalid"`;
